### ISSUE TITLE
Add a link checker for this repo

### ISSUE THREAD

**Issue by pmorie** [kubernetes/community#359]
While looking at the API conventions docs this morning, I found that many of the links in that doc were broken.  We should add a link checker for this repo (and eventually other repos in the community).

**Comment by pmorie** [comment]
@duglin any chance you might be interested in doing this?

**Comment by chupman** [comment]
I'd like to work on this. I've looked at verify-links.sh script that @duglin wrote, but I'm not totally clear on how you want the link checking implemented. Are you wanting something like a scheduled task to regularly scan for dead links? Or just something to manually run from time to time?

**Comment by duglin** [comment]
@chupman it should run as part of the verification checks that are part of the normal kube build process.  So basically it should run on every PR.  Similar to what is done for lint and gofmt checks.

**Comment by chupman** [comment]
I'm most likely overthinking it, but my concern was that if you run on every PR it could cause unrelated errors if an existing link goes dead. Would it make sense to expect an extra arg with the files changed in addition to the repo base dir? Or is it better to just leave it simple and just scan the whole repo and let the reviewer read the error log?

**Comment by duglin** [comment]
The purpose behind running the test is exact what you said... to find dead links. So we want things to fail if there area dead links.  I suspect the first time we turn this on we'll find a TON and it'll result in the PR to add this to also include a ton of md file fixes.  :-)

Now if you're concerned about having to check too many files and it being slow, we should wrapper the call to the checker with code that only calls it for files changed in the current PR/commit.

**Comment by duglin** [comment]
For reference, here the repo with the link checker:  https://github.com/duglin/vlinker

**Comment by jamiehannaford** [comment]
If nobody else is already working on this, I'd like to help out. 

@duglin are there any docs about how to set up CI on a new repo? I'm not sure what the conventions are - e.g. is it okay to use Travis or does everything need to go through the existing https://github.com/kubernetes/test-infra?

**Comment by jamiehannaford** [comment]
Also, instead of writing a new script, perhaps we could use or refactor https://github.com/kubernetes/kubernetes/blob/master/cmd/mungedocs/links.go and run all the mung scripts on a PR?

**Comment by chupman** [comment]
I'm currently waiting on code review from @duglin for some enhancements on his link checker. Despite the fact that I've already invested some time in this I agree that if there's something out there that's already maintained it's probably a better option. In case we do go with vlinker I'd also like to know about the CI integration. I was using travis on my own forks for testing while I was whittling down false positives.

**Comment by spiffxp** [comment]
https://github.com/kubernetes/kubernetes/pull/70052#discussion_r239686687
https://github.com/kubernetes/kubernetes/pull/70052#issuecomment-445112134

Turns out we _have_ a linkchecker in kubernetes/kubernetes: https://github.com/kubernetes/kubernetes/tree/master/cmd/linkcheck

It's just been broken and unused for two years.  It was in use before we heavily leaned on kubernetes/website and kubernetes/community the way we do today.  I would suggest we break that code out and see what can be done with it.

k/test-infra can be a home for it if there's no more appropriate place for it

**Comment by chupman** [comment]
PR is in for the link check, but a separate PR will be needed in this repo for the [wrapper script](https://github.com/kubernetes/kubernetes/blob/master/hack/verify-linkcheck.sh). Changes will need to be made to specify which directories should be scanned or skipped.

**Comment by chupman** [comment]
Hi @spiffxp I got the PR in (#3324) to actually add in the link checker to CI. I'm hoping someone can let me know where to save the output from the command. Once I update that it should be ready to test. We'll probably need a combination of dead link fixes and whitelisting before it can actually be merged though.

**Comment by spiffxp** [comment]
/assign
As I'm helping review PR from @chupman

**Comment by spiffxp** [comment]
/unassign
/help
@chupman wasn't able to complete https://github.com/kubernetes/community/pull/3324
